earth_enterprise/rpms/post-install_fusion.sh

```diff
@@ -0,0 +1,303 @@
+# limitations under the License.
+
+# NOTE: requires xmllint from libxml2-utils
+
+set +x
+
+#------------------------------------------------------------------------------
+
+# versions and user names
+GEE="Google Earth Enterprise"
+GEEF="$GEE Fusion"
+LONG_VERSION="5.2.1"
+ROOT_USERNAME="root"
+
+# directory locations
+BININSTALLROOTDIR="/etc/init.d"
+BININSTALLPROFILEDIR="/etc/profile.d"
+BASEINSTALLDIR_OPT="/opt/google"
+BASEINSTALLDIR_ETC="/etc/opt/google"
+BASEINSTALLDIR_VAR="/var/opt/google"
+INITSCRIPTUPDATE="/usr/sbin/update-rc.d"
+CHKCONFIG="/sbin/chkconfig"
+
+# derived directories
+SYSTEMRC="$BASEINSTALLDIR_ETC/systemrc"
+FUSIONBININSTALL="$BININSTALLROOTDIR/gefusion"
+
+#------------------------------------------------------------------------------
+# Get system info values
+NEWLINECLEANER="sed -e s:\\n::g"
+HOSTNAME="$(hostname -f | tr [A-Z] [a-z] | $NEWLINECLEANER)"
+HOSTNAME_F="$(hostname -f | $NEWLINECLEANER)"
+HOSTNAME_S="$(hostname -s | $NEWLINECLEANER)"
+HOSTNAME_A="$(hostname -a | $NEWLINECLEANER)"
+
+NUM_CPUS="$(grep processor /proc/cpuinfo | wc -l | $NEWLINECLEANER)"
+
+# config values
+ASSET_ROOT="/gevol/assets"
+SOURCE_VOLUME="/gevol/src"
+DEFAULTGEFUSIONUSER_NAME="gefusionuser"
+DEFAULTGROUPNAME="gegroup"
+GEFUSIONUSER_NAME=$DEFAULTGEFUSIONUSER_NAME
+GROUPNAME=$DEFAULTGROUPNAME
+
+# script arguments
+START_FUSION_DAEMON=true
+
+# additional variables
+IS_NEWINSTALL=false
+PUBLISH_ROOT_VOLUME=""
+
+ASSET_ROOT_VOLUME_SIZE=0
+
+MIN_ASSET_ROOT_VOLUME_SIZE_IN_KB=1048576
+EXISTING_HOST=""
+IS_SLAVE=false
+NEW_GEFUSIONUSER=false
+NEW_GEGROUP=false
+
+#-----------------------------------------------------------------
+# Main Functions
+#-----------------------------------------------------------------
+main_postinstall()
+{
+    create_system_main_directories
+
+    compare_asset_root_publishvolume
+
+    check_asset_root_volume_size
+
+    setup_fusion_daemon
+	
+    # store fusion version
+    echo $LONG_VERSION > $BASEINSTALLDIR_ETC/fusion_version
+
+    chmod 755 $BININSTALLROOTDIR/gefusion
+    chmod 755 $BININSTALLPROFILEDIR/ge-fusion.csh
+    chmod 755 $BININSTALLPROFILEDIR/ge-fusion.sh
+
+    check_fusion_master_or_slave
+
+    install_or_upgrade_asset_root
+
+    fix_postinstall_filepermissions
+    final_assetroot_configuration
+
+    final_fusion_service_configuration
+}
+
+#-----------------------------------------------------------------
+# Post-install Functions
+#-----------------------------------------------------------------
+setup_fusion_daemon()
+{
+	# setup fusion daemon
+	printf "Setting up the Fusion daemon...\n"
+
+	test -f $CHKCONFIG && $CHKCONFIG --add gefusion
+	test -f $INITSCRIPTUPDATE && $INITSCRIPTUPDATE -f gefusion remove
+	test -f $INITSCRIPTUPDATE && $INITSCRIPTUPDATE gefusion start 90 2 3 4 5 . stop 10 0 1 6 .
+	
+	printf "Fusion daemon setup ... Done\n"
+}
+
+create_system_main_directories()
+{
+        mkdir -p $ASSET_ROOT
```

Comment:
We should include these directories as part of the RPM, instead of creating them in a post-install script. This would prevent the package from being able to remove this and parent directories upon uninstall.